tests/PendingTenantsTest.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Support\Facades\Artisan;
+use Illuminate\Support\Facades\Event;
+use Stancl\Tenancy\Commands\ClearPendingTenants;
+use Stancl\Tenancy\Commands\CreatePendingTenants;
+use Stancl\Tenancy\Events\CreatingPendingTenant;
+use Stancl\Tenancy\Events\PendingTenantCreated;
+use Stancl\Tenancy\Events\PendingTenantPulled;
+use Stancl\Tenancy\Events\PullingPendingTenant;
+use Stancl\Tenancy\Tests\Etc\Tenant;
+
+test('tenants are correctly identified as pending', function (){
+    Tenant::createPending();
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+
+    Tenant::onlyPending()->first()->update([
+        'pending_since' => null
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('pending trait adds query scopes', function () {
+    Tenant::createPending();
+    Tenant::create();
+    Tenant::create();
+
+    expect(Tenant::onlyPending()->count())->toBe(1)
+        ->and(Tenant::withPending(true)->count())->toBe(3)
+        ->and(Tenant::withPending(false)->count())->toBe(2)
+        ->and(Tenant::withoutPending()->count())->toBe(2);
+
+});
+
+test('pending tenants can be created and deleted using commands', function () {
+    config(['tenancy.pending.count' => 4]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(4);
+
+    Artisan::call(ClearPendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('CreatePendingTenants command can have an older than constraint', function () {
+    config(['tenancy.pending.count' => 2]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(5)->timestamp
+    ]);
+
+    Artisan::call('tenants:pending-clear --older-than-days=2');
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+});
+
+test('CreatePendingTenants command cannot run with both time constraints', function () {
+    pest()->artisan('tenants:pending-clear --older-than-days=2 --older-than-hours=2')
+        ->assertFailed();
+});
+
+test('CreatePendingTenants commands all option overrides any config constraints', function () {
+    Tenant::createPending();
+    Tenant::createPending();
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(10)
+    ]);
+
+    config(['tenancy.pending.older_than_days' => 4]);
+
+    Artisan::call(ClearPendingTenants::class, [
+        '--all' => true
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('tenancy can check for pending tenants', function () {
```

Comment:
```suggestion
test('tenancy can accurately check if there are any pending tenants', function () {
```

Maybe this name would be a bit better? Also I'm not sure if this test is needed. I think there could just be a test that checks if `Tenant::onlyPending()` works correctly. If that method works correctly, we can assume that `->exists()` works too (as well as other query methods).

Reply:
re: name — agree on the change but get rid of the `accurately`, there's no way you could get inaccurate or semi-accurate results here. It either can check or it can't.

> I think there could just be a test that checks if

Do you mean adding a new test?

Reply:
> there's no way you could get inaccurate or semi-accurate results here. It either can check or it can't.

Yeah, you're right.

```suggestion
test('tenancy can check if there are any pending tenants', function () {
```

> Do you mean adding a new test?

 I meant we could remove this test along with some other tests and add a test that checks if `Tenant::onlyPending()` works correctly because that should be enough (I saw this test essentially as `check if eloquent's exist() method works`). But this test is the only one that we'd get rid of, so now, I don't think the idea was that great.

Reply:
We *could* remove this test since the method is being used in other tests, but if it's a super short test, we can just keep it.

Reply:
> but if it's a super short test, we can just keep it

Right, I'd keep it since it's short

Reply:
Remove the `Tenant::query()->delete();` call, update the name as suggested above, and you can mark this as resolved 👍🏻